src/main/java/uk/gov/justice/digital/domain/DomainExecutor.java

```diff
@@ -60,13 +60,21 @@ protected void insertTable(TableIdentifier info, Dataset<Row> dataFrame)
         String tablePath = storage.getTablePath(info.getBasePath(), info.getSchema(), info.getTable());
         logger.info("Domain insert to disk started");
         try {
-            if (!storage.exists(spark, info)) {
-                storage.create(tablePath, dataFrame);
-                schema.create(info, tablePath, dataFrame);
-                logger.info("Creating delta table completed...");
+            if(storage.exists(spark, info)) {
+                if(storage.hasRecords(spark, info)) {
+                    // this is trying to overwrite so throw an exception
+                    throw new DomainExecutorException("Delta table " + info.getTable() + " already exists and has records. Try replace");
+                } else {
+                    storage.replace(tablePath, dataFrame);
+                    logger.warn("Storage is empty so performing a replace instead");
+                }
             } else {
-                throw new DomainExecutorException("Delta table " + info.getTable() + "already exists");
+                storage.create(tablePath, dataFrame);
```

Comment:
create method not having an overwrite (.option("overwriteSchema", true))option. Can we test this scenario to ensure it works without the overwrite option. Instead we can call update() method in this block